### DISCUSSION TITLE
MIT 6.046: Design and Analysis of Algorithms - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#534] [Announcements]
# MIT 6.046: Design and Analysis of Algorithms - CS自学指南

CS自学指南

https://csdiy.wiki/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95/6.046/

**Comment by JSML0707** [comment]
6.854怎么样，好像是深入版本

**Reply by naiveguy114514** [reply]
里面好像说了

**Reply by sanbei101** [reply]
竟然在这里推销,傻逼去死吧

**Comment by ChengJunJunJun** [comment]
谢谢

**Comment by stormwyrmx** [comment]
大四快毕业了才看到，😂

**Comment by zzhaire** [comment]
大三考研才看到

**Reply by zhangzsy** [reply]
我也是，比你还晚

**Comment by MOSSJD** [comment]
很庆幸大一就看到了这个项目, 感谢作者与贡献者们的无私分享!

**Comment by CasearF** [comment]
大二看到了，感觉还来的及。想拉开于周围同学的差距

**Comment by MOSSJD** [comment]
加油啊


3369587019
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PKUFlyingPig/cs-self-learning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 晚上9:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PKUFlyingPig/cs-self-learning] MIT 6.046: Design and Analysis of Algorithms - CS自学指南 (Discussion #534)





 
大二看到了，感觉还来的及。想拉开于周围同学的差距

**Comment by ccckkk567** [comment]
有没有类似的人工智能学习网站捏？

**Reply by Liunique2023** [reply]
有的，是阿里和腾讯的资深人工智能研究员写的。https://www.huaxiaozhuan.com/
其实很多资源都是碍于国内墙的存在，这些因素也算是阻碍我们发展的一个，唉，不敢往后说了。

**Reply by ccckkk567** [reply]
谢谢

**Comment by Liunique2023** [comment]
谢谢

**Reply by Liunique2023** [reply]
本科是某C2的电子信息工程，课程培养烂透了，一点CS的专业课都不涉及，虽然我在大一暑假就知道了这点，奈何自己不能知行合一，现在大四想转码真的是很难。

**Reply by Liunique2023** [reply]
啊啊啊啊，怪学校也没用，作者也说了，国内大学都一个样，还得靠自己自学。

**Reply by ccckkk567** [reply]
是啊，我是一所中九的，也是因为人工智能专业国内学校都是这几年才开设的，课程设计一言难尽，虽然到我这一届已经好很多了，但是还是感觉不是很好，谢谢你的推介。

**Reply by ElysiaIWTB** [reply]
觉得课程培养烂透了但是自己却不早点行动, 然后就开始抱怨么, 你和那些蠢货最大的不同就是你至少意识到自己的行动力不足,三年过去了, 我想你已经有所进步了

**Comment by Chtkenny** [comment]
大一CS前来报到

**Comment by xiaoyvxie** [comment]
很庆幸高二就看到了这个项目

**Comment by shuibinlong** [comment]
两年半[警觉]

**Comment by x-z-y** [comment]
今天发现cs61a往年的课程网站都打不开了，但是前天看还是可以进去的，有人遇到一样的问题吗
https://inst.eecs.berkeley.edu/~cs61a/fa22/

**Reply by takagi18065** [reply]
已解决，校园网的问题，直连 .edu 失败

**Reply by x-z-y** [reply]
对的

**Comment by LilanOvO** [comment]
要上研究生了才发现这样一个宝藏，希望读完研可以按着大佬的学习路线打开计算机世界的大门

**Comment by Realyyc** [comment]
感谢！我是从MIT6.042发现新大陆的！

**Comment by jenkey2011** [comment]
工作`8`年了才看到

**Comment by yulong-ge** [comment]
推荐一款youtube字幕插件，看视频的时候帮我省了不少力气 [youtube双语字幕](https://microsoftedge.microsoft.com/addons/detail/youtube%E2%84%A2-%E5%8F%8C%E5%AD%97%E5%B9%95/kicjdgmlfepkcglkdcaalgikoaphdbbp)

**Comment by turturturturtur** [comment]
好奇这个在线电子书是什么做出来的，我也想做一个😃

**Reply by js0ny** [reply]
用的Mkdocs

**Comment by daining810** [comment]
作为一个上班族, 看到这个神仙项目, 真的太感谢了!

**Comment by aixi** [comment]
也许可以加入一些搜索引擎、向量检索相关的内容，这些主题在RAG中非常流行

**Comment by cppez** [comment]
https://ds100.org/sp24/这种网址上面的作业是不是不能看

**Reply by wsgcz** [reply]
可以从https://github.com/DS-100/sp24-student上面拉下来做

**Reply by wsgcz** [reply]
https://github.com/DS-100/sp24-student

**Comment by haer2829** [comment]
我有一个非常实在的问题，我看里面都是英文课程，大家英语水平都可以看懂吗？

**Reply by Sword4177** [reply]
都是有翻译插件的，至少如果在YouTube是可以用翻译，英语可以私下学。

**Comment by 212721** [comment]
这github订阅后发我邮箱的。。。

在 2024-02-13 16:24:10，sanbei ***@***.***> 写道：


竟然在这里推销,傻逼去死吧

**Comment by luckywangyi** [comment]
按照这个顺序学习就可以吗

**Comment by jlff** [comment]
6.006 => 6.046 => 6.854 => 6.851

**Comment by Hirazawa-Yui** [comment]
请问和这个6.046(https://www.bilibili.com/video/BV1Tb411M7FA/?spm_id_from=333.788.top_right_bar_window_custom_collection.content.click&vd_source=1c95250a43c1563ccc0581e75c744bd3)是一样的吗？会有哪些不同的地方？

**Reply by Hirazawa-Yui** [reply]
噢噢这个是6.046j，好像不一样

**Comment by lllooolllqwq** [comment]
请问大佬15年就是的公开版本了吗？
没有更新一点的版本了吗？

**Comment by kawayee** [comment]
我学习的版本是2015，辅助cs170sp2026，gpt对两门课的评价非常合适，难度在同一水平，侧重方向不同。
教学风格：6.046为理论计算机科学家做铺垫，cs170则是本科阶段的高级算法课程，6.046强调数学证明，cs170强调数学直觉，二者都遵从各自教科书的教学风格。
课程内容：前半段二者一样，后半段不同，由于cs170根据老师不同所以经常变化，详情建议问gpt。
课程难度：相当难，建议上课的时候开两个gpt窗口，一个用于分析lecture note，另一个用于卡壳的时候随时提问。note里有很多写错的地方，gpt会自动修正并提示。recitation建议以gpt为主，视频作为辅助理解（有些太过抽象的数学表述真的需要看图理解）。

PS：
1、cs170-2026引入了ai相关内容，教师是陈立杰（神秘长发男）。
2、6.042,6.006,6.046都是eric教学，你可以明显看到一个热爱马里奥的青年帅哥是怎么变成地中海大叔的。

**Reply by BruceZhangzzz** [reply]
您好能请问一下cs170sp2026网上有公开视频吗

**Reply by kawayee** [reply]
目前能找的有sp20,sp22,fa25，其余的要么借ucb学生账号，要么得有人给你油管的非公开视频的链接，我个人觉得sp22是最好的，sp20讲的比较乱，fa25印度老师口音不太能接受。
